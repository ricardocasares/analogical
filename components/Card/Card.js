// external
import React from "react";

// internal
import { themed } from "../../providers/Theme";

export default themed(({ theme, children }) => (
  <div>
    {children}
    <style jsx>{`
      div {
        padding: 2rem;
        border: 0.1rem solid ${theme.card.border};
        border-radius: 0.3rem;
        width: 100%;
        transition: border-color 0.2s;
      }

      div:hover {
        border-color: ${theme.card.hover};
      }
    `}</style>
  </div>
));
