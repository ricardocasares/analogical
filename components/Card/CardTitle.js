// external
import React from "react";

export default ({ theme, children }) => (
  <h4>
    {children}
    <style jsx>{`
      h4 {
        color: black;
        margin: 0;
        font-weight: 400;
        font-size: 1.4rem;
        padding-bottom: 15px;
        margin-bottom: 15px;
        border-bottom: 1px solid ${theme.card.border};
      }
    `}</style>
  </h4>
);
