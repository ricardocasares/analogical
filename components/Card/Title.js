// external
import React from "react";

// internal
import { themed } from "../../providers/Theme";

export default themed(({ theme, children }) => (
  <h4>
    {children}
    <style jsx>{`
      h4 {
        color: ${theme.card.title};
        margin-top: 0;
        font-weight: 400;
        font-size: 1.4rem;
      }
    `}</style>
  </h4>
));
