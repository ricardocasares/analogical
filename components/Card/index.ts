import styled from "@app/lib/styled";
import { themeGet } from "styled-system";

export const Card = styled.section`
  display: flex;
  flex-direction: column;
  padding: 15px;
  border-radius: 4px;
  border: 1px solid ${themeGet("colors.light.5")};
  transition: border-color 1s;

  a {
    transition: color 1s;
    color: ${themeGet("colors.light.2")};
  }

  h3 {
    margin: 0;
    font-size: 16px;
  }

  p {
    flex: 1;
    font-size: 14px;
    transition: color 1s;
    color: ${themeGet("colors.light.4")};
  }

  &:hover {
    border-color: ${themeGet("colors.light.1")};
    a {
      color: ${themeGet("colors.light.1")};
    }

    p {
      color: ${themeGet("colors.light.2")};
    }
  }
`;
