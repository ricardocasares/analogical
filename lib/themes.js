// external
import c from "open-color";

// internal
import { SANS, MONO } from "./fonts";

export const dark = {
  global: {
    time: c.gray[4],
    code: c.gray[6],
    text: c.gray[7],
    background: c.gray[0],
    anchors: c.gray[7],
    blockquote: c.gray[7],
    strong: c.gray[9]
  },
  logo: c.gray[9],
  pic: {
    background: c.gray[8],
    loading: [c.gray[8], c.gray[7]]
  },
  posts: {
    list: c.gray[2]
  },
  card: {
    title: c.gray[9],
    hover: c.gray[5],
    border: c.gray[3]
  },
  type: {
    title: c.gray[8],
    muted: c.gray[7],
    fonts: {
      sans: SANS,
      mono: MONO
    }
  },
  chart: {
    title: c.gray[9],
    subtitle: c.gray[6],
    colors: [
      c.pink[6],
      c.grape[6],
      c.violet[6],
      c.indigo[6],
      c.blue[6],
      c.cyan[6],
      c.teal[6],
      c.green[6],
      c.lime[6],
      c.yellow[6],
      c.orange[6],
      c.red[6]
    ]
  },
  loading: c.violet[6],
  codeblock: {
    color: c.gray[7],
    background: c.gray[2]
  },
  comment: {
    author: c.gray[4],
    border: c.gray[7]
  }
};

export const cyan = {
  loading: c.cyan[6],
  logo: c.cyan[6]
};

export const teal = {
  loading: c.teal[6],
  logo: c.teal[6]
};

export const orange = {
  loading: c.orange[6],
  logo: c.orange[6]
};

export const blue = {
  loading: c.blue[6],
  logo: c.blue[6]
};
