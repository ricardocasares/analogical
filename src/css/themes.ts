import { createTheme } from "@/css";
import { slateDark, indigoDark } from "@radix-ui/colors";

export const dark = createTheme({
  colors: {
    ...slateDark,
    ...indigoDark,
  },
});
