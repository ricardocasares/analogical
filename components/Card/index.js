import Kard from "./Card";
import Title from "./Title";
import { themed } from "../../providers/Theme";

export const Card = themed(Kard);
export const CardTitle = themed(Title);
