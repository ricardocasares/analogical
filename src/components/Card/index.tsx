import { FunctionComponent as F } from "react";
import { Box } from "@/components/Box";
import { H4, P, A } from "@/components/Typography";

export type Card = {
  id: string;
  url: string;
  name: string;
  stars: number;
  description: string;
} & Box;

export const Card: F<Card> = ({ id, url, name, description, ...props }) => (
  <Box border="1px solid #666" borderRadius="4px" p={3} {...props}>
    <H4 style={{ margin: 0 }}>
      <A href={url}>{name}</A>
    </H4>

    <P>{description}</P>
  </Box>
);
