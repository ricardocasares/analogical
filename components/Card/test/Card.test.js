import Card from "../Card";
import { dark } from "../../../lib/themes";
import renderer from "react-test-renderer";

describe("Carding", () => {
  it("Card", () => {
    const card = renderer.create(<Card theme={dark}>Hello</Card>).toJSON();

    expect(card).toMatchSnapshot();
  });
});
