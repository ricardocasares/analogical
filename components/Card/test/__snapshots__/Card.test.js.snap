// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carding Card 1`] = `
<div
  className="jsx-3259594531"
>
  Hello
</div>
`;
